content/blog/announcing-enum-support/index.md

```diff
@@ -0,0 +1,365 @@
+{{< choosable language csharp >}}
+
+![ENUM_CSHARP](csharp-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+![ENUM_GO](go-enum.gif)
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+## Optimized for flexibility
+
+While some properties make sense as "strict" enums (i.e., the input value **must** be one of the enumerated values), there are times when it would be beneficial to have constants for commonly used values without restricting the input to **only** those values. We call this second category "relaxed" enums, and we model them a little differently in our SDKs.
+
+### "Strict" enums
+
+A property is a "strict" enum when the input value **must** be one of the enumerated values. In this case, the property type is specified as the enum type.
+
+We will use "strict" enums when we are sure that the enum will include all legal values, such as when a provider is auto-generated from a specification provided by the cloud provider (like our [Azure NextGen](https://www.pulumi.com/blog/announcing-nextgen-azure-provider/) or [Kubernetes](https://www.pulumi.com/docs/intro/cloud-providers/kubernetes/#pulumi-kubernetes-provider) providers).
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+```typescript
+export class StorageAccount extends pulumi.CustomResource {
+    constructor(name: string, args: StorageAccountArgs, opts?: pulumi.CustomResourceOptions) {...}
+}
+
+export interface StorageAccountArgs {
+    readonly accessTier?: pulumi.Input<enums.storage.latest.AccessTier>;
+    ...
+}
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+class StorageAccount(pulumi.CustomResource):
+    def __init__(__self__,
+                 resource_name: str,
+                 access_tier: Optional[pulumi.Input['AccessTier']] = None,
+                ...):
+        ...
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+namespace Pulumi.AzureNextGen.Storage.Latest
+{
+    public partial class StorageAccount : Pulumi.CustomResource
+    {
+        public StorageAccount(string name, StorageAccountArgs args, ...)
+        {
+        }
+    }
+
+    public sealed class StorageAccountArgs : Pulumi.ResourceArgs
+    {
+        [Input("accessTier")]
+        public Input<Pulumi.AzureNextGen.Storage.Latest.AccessTier>? AccessTier { get; set; }
+
+        ...
+    }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+func NewStorageAccount(ctx *pulumi.Context,
+name string, args *StorageAccountArgs, ...) (*StorageAccount, error) {
+    ...
+}
+
+type StorageAccountArgs struct {
+    AccessTier AccessTier
+    ...
+}
+```
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+**Note:** When using TypeScript, "strict" enums will accept both the constant (`AccessTier.Cool`) or the literal string (`"Cool"`). If there is an accidental spelling error in the literal, you are *immediately* alerted to the issue rather than having to wait until runtime.
```

Comment:
```suggestion
{{% notes type="info" %}}
When using TypeScript, "strict" enums will accept both the constant (`AccessTier.Cool`) or the literal string (`"Cool"`). If there is an accidental spelling error in the literal, you are *immediately* alerted to the issue rather than having to wait until runtime.
{{% /notes %}}
```